src/main/java/com/internet/shop/controllers/CartController.java

```diff
@@ -0,0 +1,51 @@
+package com.internet.shop.controllers;
+
+import com.internet.shop.lib.Injector;
+import com.internet.shop.models.ShoppingCart;
+import com.internet.shop.services.ProductService;
+import com.internet.shop.services.ShoppingCartService;
+import java.io.IOException;
+import java.util.NoSuchElementException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class CartController extends HttpServlet {
+    private static final Long USER_ID = 1L;
+
+    private static final Injector injector =
+            Injector.getInstance("com.internet.shop");
+
+    private ShoppingCartService shoppingCartService =
+            (ShoppingCartService) injector.getInstance(ShoppingCartService.class);
+    private ProductService productService =
+            (ProductService) injector.getInstance(ProductService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        try {
+            ShoppingCart shoppingCart = shoppingCartService.getByUserId(USER_ID);
+            req.setAttribute("cart", shoppingCart.getProducts());
+            req.getRequestDispatcher("/WEB-INF/views/cart/cart.jsp").forward(req, resp);
+        } catch (NoSuchElementException e) {
+            ShoppingCart shoppingCart = new ShoppingCart();
+            shoppingCart.setUserId(USER_ID);
+            shoppingCartService.create(shoppingCart);
+            req.setAttribute("cart", shoppingCart.getProducts());
+            req.getRequestDispatcher("/WEB-INF/views/cart/cart.jsp").forward(req, resp);
+        }
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
```

Comment:
let's create a separate controller for adding products

Reply:
fixed